### DISCUSSION TITLE
View counter

### DISCUSSION THREAD

**Discussion by oim37** [wintercms/wn-blog-plugin#53] [Q&A] [answered]
Hello,

Maybe someone has already encountered or created such a functionality for themselves; I want to display a view counter on each page.
As I understand, the standard functionality does not have such an option.

**Comment by mjauvin** [comment]
Can you describe what you want to achieve in more details?

**Comment by oim37** [comment]
It would be interesting to see the number of views for each page, I use this module for the knowledge base and have seen such functionality on other sites and want to display it too.

The example in the screenshot is how they count them, I don’t know, the simplest method would suit me.
![image](https://github.com/wintercms/wn-blog-plugin/assets/6842168/1a5d9ef3-ffb3-4356-be43-b23de3453855)

**Comment by mjauvin** [comment, marked as answer]
Ok, so you'd like to keep a count of blog posts view (per post) and display this on the blog post page ?

If so, you could dynamically extend the blog post component or model and update/save the information in the post "metadata" db column (jsonable field).

**Comment by oim37** [comment]
Yes, that's right, but maybe there is a ready-made module that generally counts all pages, not just the blog?

**Reply by mjauvin** [reply]
Not that I'm aware of, no. I could be wrong.

**Reply by oim37** [reply]
Understood thanks :)

**Comment by mjauvin** [comment]
One way to count ALL page access would we to use the `cms.page.beforeDisplay` event (https://wintercms.com/docs/v1.2/api/events/cms/page/beforeDisplay) and save the information in a db table created by a plugin that would key using the full url as the key and another column for the count.

**Reply by oim37** [reply]
Thanks a lot
